whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/MessageService.java

```diff
@@ -53,7 +54,11 @@ public void verifyAndSetAsUndeliverable(final EmailMessageInfo message){
         }
 
         LOGGER.debug("Undeliverable message-id {} email {}", message.messageId(), StringUtils.join(message.emailAddresses(), ", "));
-        message.emailAddresses().forEach(email -> emailStatusDao.createEmailStatus(email, EmailStatus.UNDELIVERABLE));
+        try {
+            message.emailAddresses().forEach(email -> emailStatusDao.createEmailStatus(email, EmailStatus.UNDELIVERABLE));
+        } catch (DuplicateKeyException ex) {
```

Comment:
You need to handle the exception per email instead of all email , else it will skip creating other emails in list as soon as this exception is thrown